src/accelerate/logging.py

```diff
@@ -40,11 +41,26 @@ def log(self, level, msg, *args, **kwargs):
 
         Accepts a new kwarg of `main_process_only`, which will dictate whether it will be logged across all processes
         or only the main executed one. Default is `True` if not passed
+
+        Also accepts "in_order", which if `True` makes the processes log one by one, in order.
+        This is much easier to read, but comes at the cost of sometimes needing to wait for
+        the other processes. Default is `False` to not break with the previous behavior.
+
+        `in_order` is ignored if `main_process_only` is passed.
         """
         main_process_only = kwargs.pop("main_process_only", True)
+        in_order = kwargs.pop("in_order", False)
+
         if self.isEnabledFor(level) and self._should_log(main_process_only):
             msg, kwargs = self.process(msg, kwargs)
             self.logger.log(level, msg, *args, **kwargs)
+        elif self.isEnabledFor(level) and in_order:
```

Comment:
Let's refactor this under one `if self.isEnabledFor()` and then have our checks for `main_process_only` and `in_order`. Likely this check should be first and then move to the `main_process_only` (so reverse what we have here)